src/AppBundle/Controller/DashboardController.php

```diff
@@ -0,0 +1,345 @@
+use AppBundle\Entity\User;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Ob\HighchartsBundle\Highcharts\Highchart;  //Highcharts bundle
+use Symfony\Component\HttpFoundation\Request;
+use AppBundle\Entity\Component;
+use AppBundle\Form\ComponentType;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+
+
+class DashboardController extends Controller
+{
+
+    public function indexAction()
+    {
+        return $this->render('AppBundle::layout.html.twig'
+        );
+    }
+
+    //.......Dashboard view action
+    /***
+     * @param Request $request
+     * @param $id
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function viewAction(Request $request, $id)
+    {
+        //.......Get current logged user
+        $user = $this->container->get('security.context')->getToken()->getUser();
+        $userId = $user->getId();
+
+        $repository = $this
+            ->getdoctrine()
+            ->getManager()
+            ->getRepository('AppBundle:Component');
+
+        //.......Get current dashboard components
+        $components = $repository->findBy(
+            array('dashboard' => $id));
+
+        //.......Initialize chart's variables
+        $charts = array();
+        $componentName = array();
+        $componentId = array();
+        $sizeComponent = array();
+        $forms = array();
+
+        //.......For each component set chart's variables and create a form for each component
+        foreach ($components as $key => $component) {
+            $chart = $this->container->get('app.charts');
+            $chart->setLegend($component->getLegend());
+            $chart->setRequestSql($component->getRequestSQL());
+            $chart->setTypeGraph($component->getTypeGraph());
+            $chart->setXAxis($component->getXAxis());
+            $chart->setYAxis($component->getYAxis());
+            $chart = $chart->generateChart($key);
+            array_push($charts, $chart);
+            array_push($componentName, $component->getName());
+            array_push($componentId, $component->getId());
+            array_push($sizeComponent, $component->getSizeComponent());
+            $forms[$component->getId()] = $this->get('form.factory')->createNamedBuilder($component->getId(), new ComponentType(), $component)->getForm();
+        }
+        //.......Get current dashboard object
+        $em = $this->getDoctrine()->getManager();
+        $dashboard = $em->getRepository('AppBundle:Dashboard')
+            ->find($id);
+        //.......Set dashboard's creator
+        $creator = $dashboard->getCreator();
+
+        //.......Generate form for "Edit Dashboard"
+        $form = $this->get('form.factory')->create(new DashboardType(), $dashboard);
+        $form->handleRequest($request);
+
+        //.......Check if the method is POST and which form component is submitted
+        if ($request->isMethod('POST')) {
```

Comment:
Wouldn't be better to have one action to view the dashboard and another one to edit it?